src/elm/clinical.ts

```diff
@@ -2,26 +2,34 @@ import { Expression } from './expression';
 import * as dt from '../datatypes/datatypes';
 import { Context } from '../runtime/context';
 import { build } from './builder';
+import { resolveValueSet } from '../util/util';
 
 export class ValueSetDef extends Expression {
   name: string;
   id: string;
   version?: string;
+  codesystems?: CodeSystemRef[];
 
   constructor(json: any) {
     super(json);
     this.name = json.name;
     this.id = json.id;
     this.version = json.version;
+    this.codesystems = json.codeSystem?.map((cs: any) => new CodeSystemRef(cs));
   }
 
-  //todo: code systems and versions
-
   async exec(ctx: Context) {
-    const valueset =
-      (await ctx.codeService.findValueSet(this.id, this.version)) ||
-      new dt.ValueSet(this.id, this.version);
-    ctx.rootContext().set(this.name, valueset);
+    let codeSystems;
+    if (this.codesystems) {
+      codeSystems = await Promise.all(
+        this.codesystems.map(async cs => {
+          const csdef = (await cs.exec(ctx)) as CodeSystemDef;
```

Comment:
I feel like we should just execute and return the CodeSystemDef here.

Reply:
This is used to create the `CQLValueSet`, which requires an array of `CodeSystem`s

Reply:
Yes. But executing the `CodeSystemDef`s would give us the `CodeSystem` objects.

Reply:
It looks like it should actually already be executed when the CodeSystemRef is executed. I will make that change!